src/software/pipeline/main_triangulate.cpp

```diff
@@ -0,0 +1,275 @@
+#include <aliceVision/system/cmdline.hpp>
+#include <aliceVision/types.hpp>
+#include <aliceVision/config.hpp>
+#include <aliceVision/track/TracksBuilder.hpp>
+#include <aliceVision/sfm/BundleAdjustment.hpp>
+
+#include <boost/program_options.hpp>
+#include <boost/filesystem.hpp>
+
+#include <cstdlib>
+
+// These constants define the current software version.
+// They must be updated when the command line is changed.
+#define ALICEVISION_SOFTWARE_VERSION_MAJOR 2
+#define ALICEVISION_SOFTWARE_VERSION_MINOR 1
+
+using namespace aliceVision;
+
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+using namespace aliceVision::track;
+using namespace aliceVision::sfm;
+
+
+/**
+ * @brief Retrieve the view id in the sfmData from the image filename.
+ * @param[in] sfmData the SfM scene
+ * @param[in] name the image name to find (uid or filename or path)
+ * @param[out] out_viewId the id found
+ * @return if a view is found
+ */
+bool retrieveViewIdFromImageName(const sfmData::SfMData& sfmData,
+    const std::string& name,
+    IndexT& out_viewId)
+{
+    out_viewId = UndefinedIndexT;
+
+    // list views uid / filenames and find the one that correspond to the user ones
+    for (const auto& viewPair : sfmData.getViews())
+    {
+        const sfmData::View& v = *(viewPair.second.get());
+
+        if (name == std::to_string(v.getViewId()) ||
+            name == fs::path(v.getImagePath()).filename().string() ||
+            name == v.getImagePath())
+        {
+            out_viewId = v.getViewId();
+            break;
+        }
+    }
+
+    if (out_viewId == UndefinedIndexT)
+        ALICEVISION_LOG_ERROR("Can't find the given initial pair view: " << name);
+
+    return out_viewId != UndefinedIndexT;
+}
+
+
+int aliceVision_main(int argc, char** argv)
+{
+    // command-line parameters
+    std::string verboseLevel = system::EVerboseLevel_enumToString(system::Logger::getDefaultVerboseLevel());
+    std::string sfmDataFilename;
+    std::vector<std::string> featuresFolders;
+    std::vector<std::string> matchesFolders;
+    std::string outputSfM;
+
+    // user optional parameters
+    std::string extraInfoFolder;
+    std::string describerTypesName = feature::EImageDescriberType_enumToString(feature::EImageDescriberType::SIFT);
+    std::pair<std::string, std::string> initialPairString("", "");
+
+    sfm::ReconstructionEngine_sequentialSfM::Params sfmParams;
+    int maxNbMatches = 0;
+    int minNbMatches = 0;
+    bool useOnlyMatchesFromInputFolder = false;
+    bool computeStructureColor = true;
+
+    int randomSeed = std::mt19937::default_seed;
+
+    po::options_description allParams(
+        "Triangulation\n"
+        "Perform the triangulation from the incremental SfM step\n"
+        "AliceVision Triangulation");
+
+    po::options_description requiredParams("Required parameters");
+    requiredParams.add_options()
+        ("input,i", po::value<std::string>(&sfmDataFilename)->required(),
```

Comment:
It should be documented that the input SfM data must already contain the poses (as this is one of the main differences with incremental SfM)